core/src/main/java/kafka/server/share/ShareFetchUtils.java

```diff
@@ -161,6 +161,20 @@ static long offsetForLatestTimestamp(TopicIdPartition topicIdPartition, ReplicaM
         return timestampAndOffset.get().offset;
     }
 
+    /**
+     * The method is used to get the offset for the given timestamp for the topic-partition.
+     *
+     * @return The offset for the given timestamp.
+     */
+    static long offsetForTimestamp(TopicIdPartition topicIdPartition, ReplicaManager replicaManager, long timestampToSearch, int leaderEpoch) {
+        Option<FileRecords.TimestampAndOffset> timestampAndOffset = replicaManager.fetchOffsetForTimestamp(
+            topicIdPartition.topicPartition(), timestampToSearch, new Some<>(IsolationLevel.READ_UNCOMMITTED), Optional.of(leaderEpoch), true).timestampAndOffsetOpt();
+        if (timestampAndOffset.isEmpty()) {
+            throw new OffsetNotAvailableException("offset for timestamp " + timestampToSearch + " not found for topic partition: " + topicIdPartition);
```

Comment:
nit: Please capitalise the "Offset" for consistency with the `offsetForEarliestTimestamp` and `offsetForLatestTimestamp` methods. I know this is a trivial point but we have high standards for these things.

Reply:
Thank you very much for your careful review! 😁